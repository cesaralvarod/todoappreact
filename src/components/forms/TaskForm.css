form {
  width: 100%;
  min-width: 100%;
  margin-top: 2rem;
}

form input[type="text"] {
  min-width: 100%;
  max-width: 100%;
  border: none;
  background: none;
  border: 2px solid #ccc;
  color: #000;
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  font-size: 1.6rem;
  padding: 1rem;
  outline: none;
}

form input[type="text"]:focus {
  outline: none;
}

form input[type="text"]:focus-within {
  border: 2px solid #deab12;
}

form .container-buttons {
  display: flex;
  justify-content: center;
  gap: 1rem;
  margin-top: 1rem;
}

form button {
  border: none;
  background: none;
  background: #45b818;
  border-radius: 0.2rem;
  color: #fff;
  padding: 0.5rem 0.8rem;
  font-size: 2rem;
  cursor: pointer;
}

form button:hover {
  background: #54d124;
}
