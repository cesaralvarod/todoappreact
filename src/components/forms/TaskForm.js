import React, { useRef, useEffect, useContext } from "react";
import TaskContext from "../../helpers/TaskContext";

// Styles

import "./TaskForm.css";

const TaskForm = () => {
  const { setArrData } = useContext(TaskContext);

  const inputRef = useRef(null);

  useEffect(() => {
    inputRef.current.focus();
  }, []);

  const handleSubmit = (event) => {
    event.preventDefault();

    const { value } = inputRef.current;

    console.log(value);

    if (value === "") {
      return;
    }

    setArrData({
      type: "add",
      dataForm: { id: Date.now(), content: value.trim(), done: false },
    });

    event.target.reset();
  };

  return (
    <form onSubmit={handleSubmit}>
      <input ref={inputRef} type="text" />
      <div className="container-buttons">
        <button>Add</button>
      </div>
    </form>
  );
};

export default TaskForm;
