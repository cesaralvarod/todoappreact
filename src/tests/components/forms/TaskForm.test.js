import React from "react";
import { mount } from "enzyme";
import TaskForm from "../../../components/forms/TaskForm";
import TaskContext from "../../../helpers/TaskContext";
import userEvent from "@testing-library/user-event";

describe("Testing in component form TaskForm", () => {
  const setArrData = jest.fn();

  const TestComponent = () => (
    <TaskContext.Provider value={{ setArrData }}>
      <TaskForm />
    </TaskContext.Provider>
  );

  const wrapper = mount(<TestComponent />);

  test("Component form TaskForm should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Component TaskForm shouldn't call sendArrData() 0 times", () => {
    const formSubmit = wrapper.find("form").prop("onSubmit");

    const event = { preventDefault() {}, target: "" };

    formSubmit(event);

    expect(setArrData).toHaveBeenCalledTimes(0);
  });

  test("Component Taskshould call sendArrData() one time", () => {
    const useRefSpy = jest
      .spyOn(React, "useRef")
      .mockReturnValueOnce({ current: document.createElement("button") });

    const wrapper = mount(<TestComponent />);

    const inputTask = wrapper.find("form input");

    const formSubmit = wrapper.find("form").prop("onSubmit");

    const event = { preventDefault() {}, target: "" };

    formSubmit(event);

    // expect(setArrData).toHaveBeenCalledTimes(1);
  });
});
