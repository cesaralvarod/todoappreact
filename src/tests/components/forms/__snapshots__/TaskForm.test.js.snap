// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in component form TaskForm Component form TaskForm should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    type="text"
  />
  <div
    className="container-buttons"
  >
    <button>
      Add
    </button>
  </div>
</form>
`;
